src/components/UTIL_FormField.component

```diff
@@ -0,0 +1,113 @@
+            <div class="{!'slds-form-element__control' + IF($ObjectType[sObjType].fields[field].type!='boolean', ' slds-has-divider--bottom slds-p-left--small','')}">
+                <!-- Blank space fixes issue where an empty span on read only fields were blank causing the line to shift when no value present -->
+                <apex:outputField id="outputX" value="{!sObj[field]}" styleClass="slds-form-element__static" rendered="{!ftype!='boolean'}">&nbsp;</apex:outputField>
+                <apex:outputPanel layout="{!IF(ftype=='boolean', 'inline', 'none')}" styleClass="{!IF(ftype=='boolean', 'slds-checkbox', '')}">
+                    <apex:outputLabel for="outputCB" styleClass="slds-checkbox__label" rendered="{!ftype=='boolean'}">
+                        <apex:inputCheckbox value="{!sObj[field]}" id="outputCB" disabled="true" rendered="{!ftype=='boolean'}" />
+                        <span class="slds-checkbox--faux"></span>
+                        <span class="slds-form-element__label">{!label}</span>
+                    </apex:outputLabel>
+                </apex:outputPanel>
+            </div>
+        </div>
+    </apex:outputPanel>
+
+    <!-- Updateable fields will render as input fields with the appropriate slds style - note that Record Type Id is displayed as a picklist by Salesforce. -->
+    <apex:outputPanel layout="block" styleClass="{!'slds-form-element ' + styleClass + fixup}" id="divInput"
+        rendered="{!($ObjectType[sObjType].fields[field].updateable
+                    || $ObjectType[sObjType].fields[field].createable && sObj['Id']=='') && $ObjectType[sObjType].fields[field].type != 'boolean'}">        
+        <apex:outputLabel for="inputX" styleClass="slds-form-element__label" rendered="{!ftype!='boolean'}">
+            <apex:outputText rendered="{!required}">
+                <abbr class="slds-required" title="{!$Label.lblRequired}">*</abbr>
+            </apex:outputText>{!label}
+        </apex:outputLabel>
+        <div class="slds-form-element__control">
+            <apex:inputField id="inputX" required="{!required}" value="{!sObj[field]}" label=""
+                styleClass="{!CASE(ftype,
+                                        'textarea', 'slds-textarea',
+                                        'picklist', 'slds-select',
+                                        'multipicklist', 'slds-select multipicklistSLDSFixup',
+                                        'reference', IF(field=='RecordTypeId', 'slds-select', 'lookupInput ' + field),
+                                        'slds-input')}">
+                <apex:actionSupport event="onchange" action="{!actSupAction}" reRender="{!actSupReRender}" rendered="{!actSup}" status="{!actSupStatus}" immediate="{!actSupImmediate}"/>
+            </apex:inputField>
+        </div>
+        <!-- Javascript will add the onchange event when needed and swap out the icon for lookup fields -->
+        <script type="text/javascript">
```

Comment:
since this javascript will get included onto the page multiple times, do we have any issues with javascript conflicts?  is there an official way in javascript to encapsulate this code so it won't interfere with other code?  I'm not a javascript expert, so I don't know if this is a problem or not.